### PR TITLE
New setting to disable keeping raw request body in RAM

### PR DESCRIPTION
**Pull request by bigpresh**
Right.  Back in PR #1134, I stopped us from storing the raw request body in RAM (in the `Dancer::Request` object's `body` attribute) to avoid massive RAM usage for large requests - e.g. large file uploads, etc.

It replaced `request->body` with a convenience method that, if called, grabbed the filehandle that `HTTP::Body->body` gives you, read the content from it and returned it.

However, as per issue #1140, that caused unpleasant surprises - as `HTTP::Body` only stores the body into a filehandle in some cases, not all.

In #1153 I worked on replacing our use of HTTP::Body with a bundled version with some changes to make it more predictable - but I don't really want us to have to do that, and it still fell short of sanity for some cases, e.g. multipart file uploads.

I think this is the best solution - revert out the previous stuff, and store the request body in RAM as per the expected behaviour before #1134 (pre 1.3143) by default, to avoid surprises, but add a new config setting, `raw_request_body_in_ram`, which controls whether or not we keep the raw (unparsed) request body in `Dancer::Request->body` or not.  Users who want to avoid excessive RAM usage for large requests, and do not need access to the raw request body, should be able to set this to a false value.

### CODE REVIEW

File: dist.ini [hunk 1]
```diff
@@ -22,7 +22,10 @@ authority=cpan:SUKRIA
 
 [Prereqs / RuntimeRecommends ]
 YAML = 0
+<<<<<<< HEAD
```
Comment (cromedome) [inline dist.ini]: Need to resolve this merge conflict.

File: dist.ini [hunk 2]
```diff
@@ -31,9 +34,13 @@ YAML::XS = 0
 JSON = 2.90
 HTTP::Tiny = 0.014 ; for get/post/post_form
 HTTP::CookieJar = 0.008
+<<<<<<< HEAD
```
Comment (cromedome) [inline dist.ini]: This one too.

**Comment by cromedome** [comment]
I don't see anything wrong aside from the merge conflict in `dist.ini`. I'd like to poke at it a bit more, and will probably run it on one of my test servers at work for a day or two. I think I'll be ok to :+1: this though.

**Comment by bigpresh** [comment]
Finally found time for this.  If it passes its tests, I hope to merge it and get a dev release out tomorrow, give it a whirl, then get a new D1 stable release out in the coming week, as it's way overdue, and I've been sucking at taking care of D1.

**Comment by bigpresh** [comment]
Curious, I don't get the failure about trying to add t/00_compile.t multiple times locally.

**Pull request merged**